internal/shellgen/tmpl/glibc-patch.nix.tmpl

```diff
@@ -34,9 +33,19 @@
           }
         else null) args;
 
+      selectDefaultOutputs = drv: selectOutputs drv (drv.meta.outputsToInstall or [ drv.out ]);
+      selectAllOutputs = drv: drv.all;
+      selectOutputs = drv: builtins.map (output: drv.${output});
```

Comment:
how are these used?

Reply:
I can comment with some examples. They're functions that take a derivation and a list of output names, and return those outputs. So

```
selectOutputs nixpkgs#foo [ "out", "lib", "dev" ]
```

will return ` [ nixpkgs#foo.out nixpkgs#foo.lib nixpkgs#foo.dev ]`.

Reply:
Thery're used by the template to install multiple outputs from a package. It basically generates `fmt.Sprintf("selectOutputs %s %q", ..., installable.SplitOutputs())`.